src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,55 @@
+package duke;
+
+import java.time.LocalDate;
+
+import duke.command.ByeCommand;
+import duke.command.Command;
+import duke.command.DeadlineCommand;
+import duke.command.DeleteCommand;
+import duke.command.EventCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.command.MarkCommand;
+import duke.command.TodoCommand;
+import duke.command.UnmarkCommand;
+
+/**
+ * Parses the user input and executes the corresponding commands.
+ */
+public class Parser {
+    /**
+     * Returns Command object based on user's input.
+     * @param fullCommand user's input.
+     * @return specific command.
+     * @throws DukeException If command does not exist.
+     */
+    public static Command parse(String fullCommand) throws DukeException {
+        String[] commandSegments = fullCommand.split(" ", 2);
+        String mainCommand = commandSegments[0].toLowerCase().trim();
```

Comment:
Clean way of splitting fullCommand. Helps with simpler code in switch statements:)